File: Binary Search/Search a 2D Matrix II/solution.ts
function searchMatrix(matrix: number[][], target: number): boolean {
    let minCol = 0
    let maxRow = matrix.length - 1
    while(maxRow >= 0 && minCol <= matrix[0].length - 1) {
        if(target < matrix[maxRow][minCol]) {
            maxRow--
            continue
        }
        if(target > matrix[maxRow][minCol]) {
            minCol++
            continue
        }
        if(target === matrix[maxRow][minCol]) {
            return true
        }
        
    }
    return false
};

File: Binary Search/Search a 2D Matrix II/solution.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m, n = len(matrix), len(matrix[0])
        
        start = 0
        end = m*n-1
        
        while start < end:
            mid = start + (end - start) // 2
            if matrix[mid//n][mid%n] < target:
                start = mid + 1
            else:
                end = mid
        return matrix[start//n][start%n] == target
